test/hotspot/jtreg/vmTestbase/gc/gctests/ReferencesGC/ReferencesGC.java

```diff
@@ -112,21 +112,23 @@ public void run() {
                 }
 
                 for (int i = 0; i < RANGE; i++) {
-                    if (wr[i].isEnqueued()) {
-                        ++totalQ;
+                    if (wr[i].refersTo(holder[i])) {
+                        ++totalLive;
                     }
-                    if (sr[i].isEnqueued()) {
-                        ++totalQ;
+                    if (sr[i].refersTo(holder[i])) {
+                        ++totalLive;
                     }
-                    if (phr[i].isEnqueued()) {
-                        ++totalQ;
+                    if (phr[i].refersTo(holder[i])) {
+                        ++totalLive;
                     }
                 }
-                if (totalQ != 0) {
-                    throw new TestFailure("There are " + totalQ + " references in the queue instead 0 before null-assigment.");
+                if (totalLive != 3 * RANGE) {
+                    throw new TestFailure("There are " + (3 * RANGE - totalLive) + " references cleared before null-assigment.");
                 }
 
-                for (int i = 0; i < (int) (RANGE * RATIO); i++) {
+                int REMOVE = (int) (RANGE * RATIO);
```

Comment:
These two constants could be factored out as static finals to match the casing.

Reply:
I'm making REMOVE and RETAIN statics, near RANGE and RATIO.  (Meant to do that before, but forgot.)  They can't be final though, because RANGE and RATIO aren't final, and can be set from command line arguments.  So they'll get initialized in parseArgs.